src/main/java/org/springframework/retry/annotation/RecoverAnnotationRecoveryHandler.java

```diff
@@ -81,25 +83,42 @@ public T recover(Object[] args, Throwable cause) {
 	}
 
 	private Method findClosestMatch(Object[] args, Class<? extends Throwable> cause) {
-		int min = Integer.MAX_VALUE;
 		Method result = null;
-		for (Method method : this.methods.keySet()) {
-			SimpleMetadata meta = this.methods.get(method);
-			Class<? extends Throwable> type = meta.getType();
-			if (type == null) {
-				type = Throwable.class;
+
+		boolean shouldUseRecoverMethodName = !this.recoverMethodName.equals("");
+		if (shouldUseRecoverMethodName) {
```

Comment:
We don't need this variable. The `StringUtils.isEmpty()` is or friend in this case.

Reply:
Agree. Fixed.